src/huggingface_hub/file_download.py

```diff
@@ -957,6 +957,9 @@ def hf_hub_download(
             # between the HEAD and the GET (unlikely, but hey).
             if 300 <= r.status_code <= 399:
                 url_to_download = r.headers["Location"]
+                if "lfs.huggingface.co" in url_to_download or "lfs-staging.huggingface.co" in url_to_download:
+                    # Remove authorization header when downloading a LFS blob
+                    headers.pop("authorization", None)
```

Comment:
Had to remove the `authorization` header when requesting a LFS file, otherwise S3 throws a 400 with:
```xml
<Error>
   <Code>InvalidArgument</Code>
   <Message>Unsupported Authorization Type</Message>
   <ArgumentName>Authorization</ArgumentName>
   <ArgumentValue>Bearer XXX</ArgumentValue>
   <RequestId>...</RequestId> 
  <HostId>...</HostId>
</Error>
```

Not sure if this is new ? cc @julien-c @huggingface/moon-landing-back

Reply:
related to the CDN changes I've made for filename